File: VB/XPO_XtraScheduler_Simple_Example/XPAppointment.vb
Imports DevExpress.Xpo

Namespace XPO_XtraScheduler_Simple_Example

'#Region "#xpappointment"
    ' XP object
    Public Class XPAppointment
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public AllDay As Boolean ' Appointment.AllDay

        <Size(SizeAttribute.Unlimited)>
        Public Description As String ' Appointment.Description

        Public Finish As Date ' Appointment.End

        Public Label As Integer ' Appointment.Label

        Public Location As String ' Appointment.Location

        <Size(SizeAttribute.Unlimited)>
        Public Recurrence As String ' Appointment.RecurrenceInfo

        <Size(SizeAttribute.Unlimited)>
        Public Reminder As String ' Appointment.ReminderInfo

        Public Created As Date ' Appointment.Start

        Public Status As Integer ' Appointment.Status

        <Size(SizeAttribute.Unlimited)>
        Public Subject As String ' Appointment.Subject

        Public AppointmentType As Integer ' Appointment.Type

        Public Resource As XPResource
    End Class
'#End Region  ' #xpappointment
End Namespace

File: CS/XPO_XtraScheduler_Simple_Example/XPAppointment.cs
using DevExpress.Xpo;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Xml;
using System;

namespace XPO_XtraScheduler_Simple_Example {
    #region #xpappointment
    // XP object
    public class XPAppointment : XPObject {
        public XPAppointment(Session session) : base(session) { }

        public bool AllDay;              // Appointment.AllDay

        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Description;       // Appointment.Description

        public DateTime Finish;          // Appointment.End
        public int Label;                // Appointment.Label
        public string Location;          // Appointment.Location

        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Recurrence;        // Appointment.RecurrenceInfo

        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Reminder;          // Appointment.ReminderInfo

        public DateTime Created;         // Appointment.Start
        public int Status;               // Appointment.Status
        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Subject;           // Appointment.Subject
        public int AppointmentType;      // Appointment.Type
        public XPResource Resource;
    }
    #endregion #xpappointment
}